packages/typespec-client-generator-core/test/packages/paged-operation.test.ts

```diff
@@ -203,9 +231,308 @@ describe("typespec-client-generator-core: paged operation", () => {
+      strictEqual(method.operation.responses[0].type?.kind, "model");
+      strictEqual(
+        method.continuationTokenResponseSegments?.[0],
+        method.operation.responses[0].type.properties[1],
+      );
+    });
+
+    it("continuation token in response header and query parameter", async () => {
+      await runner.compileWithBuiltInService(`
+        @list
+        op test(@continuationToken @query token?: string): ListTestResult;
+        model ListTestResult {
+          @pageItems
+          items: Test[];
+          @continuationToken
+          @header
+          nextToken?: string;
+        }
+        model Test {
+          prop: string;
+        }
+      `);
+      const sdkPackage = runner.context.sdkPackage;
+      const method = getServiceMethodOfClient(sdkPackage);
+      strictEqual(method.name, "test");
+      strictEqual(method.kind, "paging");
+      strictEqual(method.continuationTokenParameterSegments?.length, 1);
+      strictEqual(method.continuationTokenParameterSegments?.[0], method.parameters[0]);
+      strictEqual(method.continuationTokenResponseSegments?.length, 1);
+      strictEqual(method.operation.responses[0].type?.kind, "model");
+      strictEqual(
+        method.continuationTokenResponseSegments?.[0],
+        method.operation.responses[0].headers[0],
+      );
+    });
+
+    it("continuation token in response header and header parameter", async () => {
+      await runner.compileWithBuiltInService(`
+        @list
+        op test(@continuationToken @header token?: string): ListTestResult;
+        model ListTestResult {
+          @pageItems
+          items: Test[];
+          @continuationToken
+          @header
+          nextToken?: string;
+        }
+        model Test {
+          prop: string;
+        }
+      `);
+      const sdkPackage = runner.context.sdkPackage;
+      const method = getServiceMethodOfClient(sdkPackage);
+      strictEqual(method.name, "test");
+      strictEqual(method.kind, "paging");
+      strictEqual(method.continuationTokenParameterSegments?.length, 1);
+      strictEqual(method.continuationTokenParameterSegments?.[0], method.parameters[0]);
+      strictEqual(method.continuationTokenResponseSegments?.length, 1);
+      strictEqual(method.operation.responses[0].type?.kind, "model");
+      strictEqual(
+        method.continuationTokenResponseSegments?.[0],
+        method.operation.responses[0].headers[0],
+      );
+    });
+
+    it("continuation token in response header and body parameter", async () => {
+      await runner.compileWithBuiltInService(`
+        @list
+        op test(@continuationToken token?: string): ListTestResult;
+        model ListTestResult {
+          @pageItems
+          items: Test[];
+          @continuationToken
+          @header
+          nextToken?: string;
+        }
+        model Test {
+          prop: string;
+        }
+      `);
+      const sdkPackage = runner.context.sdkPackage;
```

Comment:
![image](https://github.com/user-attachments/assets/ca914093-6e55-4de5-917c-2105ba89e1e6)
According to [openapi emitter](https://cadlplayground.z22.web.core.windows.net/cadl-azure/?c=aW1wb3J0ICJAdHlwZXNwZWMvdmVyc2lvbmluZyI7CtIfcmVzdNUZaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5SZXN0O9AVSHR0cNEVVsl5OwoKQHNlcnZpY2Uoe3RpdGxlOiAiVGV4dCBhdXRob3LkAJx9KQpuYW1lc3BhY2UgxBxByBs7Cgptb2RlbCBQZXQgewogIMQoOiBzdMYdfQoKQGxpc3Qgb3AgxAhQZXRzKEBjb250aW51YXRpb25Ub2tlbiB0xAY%2FyDgpOsVLQHBhZ2VJdGVtcyBwZXRzOsRkW107xBrSQm5leHTFCslGOwp9Owo%3D&e=%40typespec%2Fopenapi3&options=%7B%7D), HTTP method shall be `post` instead of `get`.